src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -75,6 +75,12 @@ public function finishView(FormView $view, FormInterface $form, array $options)
         if ($form->getConfig()->hasAttribute('prototype') && $view->vars['prototype']->vars['multipart']) {
             $view->vars['multipart'] = true;
         }
+
+        $prefixOffset = isset($options['entry_options']['block_prefix']) ? -2 : -1;
```

Comment:
The `block_prefix` could be defined in the entry type as well (custom form types with predefined custom block prefix) I'd suggest to check the option this way to cover both variants:
```suggestion
        $prefixOffset = -1;
        foreach ($form as $entryForm) {
            if ($entryForm->getConfig()->getOption('block_prefix')) {
                $prefixOffset = -2;
                break;
            }
        }
```
?

Reply:
Good call. Actually both cases might happen, the first overriding the second. I've updated the code and the tests accordingly.